File: 2026-09/src/day_05/leet_704/Solution.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        def binarySearch(nums, target):
            lo, hi = 0, len(nums) - 1

            while lo <= hi:
                mid = (lo + hi) // 2

                if nums[mid] < target:
                    lo = mid + 1
                elif nums[mid] > target:
                    hi = mid - 1
                else:
                    return mid

            return -1

        return binarySearch(nums, target)

File: 2026-09/src/day_05/leet_704/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
   public:
    int binarySearch(vector<int>& v, int target) {
        int lo = 0;
        int hi = v.size() - 1;

        while (lo <= hi) {
            int mid = (lo + hi) / 2;

            if (v[mid] < target) {
                lo = mid + 1;
            } else if (v[mid] > target) {
                hi = mid - 1;
            } else {
                return mid;
            }
        }

        return -1;
    }

    int search(vector<int>& nums, int target) {
        return binarySearch(nums, target);
    }
};
